<?xml version="1.0"?>





<robot name="ramie">

<material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>


  


<link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.6" radius="0.2"/>
      </geometry>
    </visual>
  </link>

<link name="joint1">
    <visual>
      <geometry>
        <cylinder length="0.2" radius="0.1"/>
      </geometry>
	
       <material name="blue"/>
    </visual>
  </link>

<link name="box1">
    <visual>
      <geometry>
        <box size="0.6 0.1 0.2"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.3 0 0"/>
       <material name="blue"/>
    </visual>
  </link>


<link name="joint2">
    <visual>
      <geometry>
        <cylinder length="0.2" radius="0.1"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>

<link name="box2">
    <visual>
      <geometry>
        <box size="0.6 0.1 0.2"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.3 0 0"/>
      <material name="white"/>
    </visual>
  </link>

<link name="gripper">
    <visual>
      <geometry>
        <box size="0.1 0.1 0.2"/>
      </geometry>
      <origin rpy="0 0 3.14" xyz="0 0 -0.1"/>
      <material name="white"/>
    </visual>
  </link>

<joint name="base_to_joint1" type="revolute">
    <parent link="base_link"/>
    <child link="joint1"/>
    <limit effort="1000.0" lower="-1" upper="1" velocity="0.5"/>
    <axis xyz ="0 0 1"/>
    <origin xyz="0 0 0.4"/>
  </joint>

<joint name="joint1_to_box1" type="fixed">
 <parent link="joint1"/><child link="box1"/>
 <axis xyz ="0 0 1"/>
 <origin xyz="0 0 0"/>
</joint>


  <joint name="box1_to_joint2" type="revolute">
    <parent link="box1"/>
    <child link="joint2"/>
    <limit effort="1000.0" lower="-1" upper="1" velocity="0.5"/>
    <axis xyz ="0 0 1"/>
    <origin xyz="0.60 0 0"/>
  </joint>

<joint name="joint2_to_box2" type="fixed">
 <parent link="joint2"/><child link="box2"/>

 <axis xyz ="0 0 1"/>
 <origin xyz="0 0 0"/>
</joint>

<joint name="box2_to_gripper_prismatic" type="prismatic">
 <parent link="box2"/><child link="gripper"/>
   <limit effort="1000.0" lower="-0.1" upper="0.1" velocity="0.5"/>
 <axis xyz ="0 0 1"/>
 <origin xyz="0.55 0 -0.2" rpy="3.14 0 0"/>
</joint>




</robot>
